src/components/editor/EditorArticleController.jsx

```diff
@@ -0,0 +1,127 @@
+    event.preventDefault();
+  }
+
+  mainFormChanged(event) {
+    if (event.target.name === "addAuthor") {
+      return;
+    }
+
+    let changedFlag = this.handleMainFormChange();
+
+    let newChangedState = changedFlag ? "changed" : "none";
+    if (newChangedState !== this.state.changed) {
+      this.safeSetState({
+        changed: newChangedState
+      });
+    }
+  }
+
+  authorListChanged(authorsAdded, authorsDeleted) {
+    let changedFlag = Object.keys(authorsAdded).length > 0 || 
+      _.some(authorsDeleted, (val) => {
+        return val;
+    });
+    if (!changedFlag) {
+      changedFlag = changedFlag || this.handleMainFormChange();
+    }
+
+    let newChangedState = changedFlag ? "changed" : "none";
+    if (newChangedState !== this.state.changed) {
+      this.safeSetState({
+        changed: newChangedState
+      });
+    }
+  }
+
+  handleMainFormChange() {
+    let formNode = this.formNode;
+    let fields = ["category", "issue", "description"];
+    let falcorData = this.state.data.articlesBySlug[this.props.params.slug];
+
+    return fields.some((field) => {
+      let formValue = formNode[field].value;
+      if (field === "issue") formValue = parseInt(formValue);
+      let falcorValue = falcorData[field];
+
+      if (formValue !== falcorValue) {
+        return true;
+      }
+      return false;
+    });
+  }
+
+  bindFormNodeRef(ref) {
+    this.formNode = ref;
+  }
+
+  render() {
+    if (this.state.ready) {
+      if (!this.state.data) {
+        return <div><p>No articles match this slug</p></div>;
+      }
+
+      let slug = this.props.params.slug;
+      let data = this.state.data.articlesBySlug[slug];
+
+// The input type="number" behaves badly in Chrome, TODO: change to input type="text" and do validation yourself
+      return (
+        <div>
+          <h3>{data.title}</h3>
+          <p>Change the information for the article and press Save Changes to confirm the changes.</p>
+          <br/>
+          <form onChange={this.mainFormChanged} onSubmit={this.handleSaveChanges} ref={this.bindFormNodeRef.bind(this)}>
+            Change the Category:<br/>
+            <select defaultValue={data.category} name="category">
+              {categories.map((category) => {
+                return <option value={category} key={category}>{category}</option>;
+              })}
+            </select>
+            <div style={{height: "5px"}} />
+            Change Issue Number:<br/>
+            <input type="number" defaultValue={data.issue} name="issue" />
+            <br/>
+            Change Description:<br/>
+            <textarea style={{width: "40%"}} defaultValue={data.description} name="description" />
+            <br/>
+            Edit Authors:<br/>
+            <EditAuthorList data={_.map(data.authors, (val) => {return val;})} onChange={this.authorListChanged} />
+            <br/>
+            <input type="submit" value="Save Changes" disabled={this.state.changed === "none"} />
```

Comment:
Ahh okay, I guess the reason I misunderstood you was that I thought I had already written something like that? Maybe I hadn't doesn't look like in the outdated diff anyway. But right now I have the button disabled to be `this.state.changed !== "changed"` anyway, so it'll be disabled during none, saving and in the buffer zone when it says saved until it returns to "no changes"